### ISSUE TITLE
[Feature Request] Lock reader orientation with rotation independent of browsing orientation

### ISSUE THREAD

**Issue by s0alqasim1d** [tachiyomiorg/tachiyomi#4805]
## Why/User Benefit/User Problem

- Before [142fc0e4a61656a4d608f002c1d3c72e6426fc1c] we were able to set reader mode to "forced (landscape/portrait)"  and enjoy rotation in that mode. We could still browse in either.

- However, since that commit, if we forced the rotation to landscape or portrait, the sensor is disabled and we couldn't rotate. There is a "lock" mode though and even though we could solve the rotation aspect, we needed to browse in the mode we wanted to read in which is a little annoying on devices that are not very wide.

## What/Requirements

I'm not familiar with kotlin or android, so I don't really understand much. 

If the `flag` determines whether or not the sensor is active, why does it need me to be in landscape before starting read mode in landscape? 
Since the `flag` is the old one for forced orientation with sensor why did it work then but not now? is it due to the two lines for LOCKED_${orientation} having the same `prefValue` and then showing up as one option in settings?

* I was gonna suggest adding two more options but now I'm a little confused. 
### Edit
I didn't look at the entire file only the diff. Now I see we look at the current orientation before we determine which one to lock. Still I don't think I'm better at solving this than you. 

```kotlin
    companion object {
        fun fromPreference(preference: Int, resources: Resources): OrientationType = when (preference) {
            2 -> {
                val currentOrientation = resources.configuration.orientation
                if (currentOrientation == Configuration.ORIENTATION_PORTRAIT) {
                    LOCKED_PORTRAIT
                } else {
                    LOCKED_LANDSCAPE
                }
            }
            3 -> PORTRAIT
            4 -> LANDSCAPE
            else -> FREE
        }
```
* Maybe make another two options for lock so that the first locks the current orientation, the second locks portrait and the third locks landscape. 

```kotlin
    companion object {
        fun fromPreference(preference: Int, resources: Resources): OrientationType = when (preference) {
            2 -> {
                val currentOrientation = resources.configuration.orientation
                if (currentOrientation == Configuration.ORIENTATION_PORTRAIT) {
                    LOCKED_PORTRAIT
                } else {
                    LOCKED_LANDSCAPE
                }
            }
            3 -> LOCKED_PORTRAIT // Maybe like this
            4 -> LOCKED_LANDSCAPE // Maybe like this
            5 -> PORTRAIT
            6 -> LANDSCAPE
            else -> FREE
        }
```
Am I on the right way here @arkon?

### Edit 2
Maybe instead of two more options we use another setting under reader -> display settings with values (Auto, Portrait, Landscape) and these determine the orientation to get locked in the logic in the function `fromPreference`

**Comment by arkon** [comment]
They were split out in e8d8621f06751a6aa099931cba1d8f4e4f28ddb6

**Comment by s0alqasim1d** [comment]
> They were split out in [e8d8621](https://github.com/tachiyomiorg/tachiyomi/commit/e8d8621f06751a6aa099931cba1d8f4e4f28ddb6)

Thanks @arkon.
